homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,155 @@
+"""Support for the WatchYourLAN service."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="online_status",
+        name="Online Status",
+        icon="mdi:lan-connect",
+    ),
+    SensorEntityDescription(
+        key="ip_address",
+        name="IP Address",
+        icon="mdi:ip-network",
+    ),
+    SensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        icon="mdi:lan",
+    ),
+    SensorEntityDescription(
+        key="iface",
+        name="Network Interface",
+        icon="mdi:ethernet",
+    ),
+]
+
+
+# Utility function for setting up device info consistently
+def build_device_info(device: dict[str, Any], domain: str) -> DeviceInfo:
+    """Build device info helper function."""
+    return DeviceInfo(
```

Comment:
Do we have the Mac?

Reply:
As far as I know, we should always have the MAC. Otherwise, I'm not sure how WatchYourLAN is able to ping the IP.

Reply:
Let's use `connections` instead of identifiers, and have the Mac address registered